src/igz_controls/components/validating-input-field/validating-input-field.component.js

```diff
@@ -354,12 +352,17 @@
 
             if (event.target === validationIcon) {
                 if (!lodash.isEmpty(ctrl.validationRules)) {
+                    ctrl.isValidationPopUpShown = !ctrl.isValidationPopUpShown;
+
                     $timeout(function () {
                         $element.find('.field')[0].focus();
-
                         ctrl.inputFocused = true;
-                        ctrl.isValidationPopUpShown = !ctrl.isValidationPopUpShown;
-                    });
+                        var popUpRect = $element.find('.validation-pop-up-wrapper')[0].getBoundingClientRect();
+
+                        angular.element('.validation-pop-up-wrapper').css({
+                            'height': lodash.get(popUpRect, 'height', 0) > 0 ? popUpRect.height : 'auto'
+                        });
```

Comment:
I know we do this all the time in our code base, but still:
When you use `$element` you get a jQuery object.
So `$elemet.find` also returns a jQuery object. 
Then we usually do `[0]` so we get the raw DOM element, which makes us _lose_ all the jQuery enhancements and use raw DOM API (like `getBoundingRect()`).
Now, I have nothing against the DOM API. I think it's important to be familiar with it and use it sometimes.
But if we already get jQuery for free with AngularJs, why not use it? It sometimes makes code shorter or nicer for us.

For example you use the jQuery method `.css()` to set the CSS `height` rule.
For that you use `angualr.element` to get a jQuery object.
But you already got that object on the previous line, and lost it (with `[0]`).
Instead we can store it and reuse it.
For example:
```js
var popUp = $element.find('.validation-pop-up-wrapper');

popUp.css(…);
```
Actually jQuery allows for chaining so we can even spare the variable and do:
```js
$element.find('.validation-pop-up-wrapper').css(…);
```
But because we need to reference the element again inside the parameter we pass to the `.css()` method (to get its height), we _do_ need to use a variable.
Looking here https://api.jquery.com/category/dimensions/
We see we have `.height()`, `.innerHeight()` and `.outerHeight()` methods.
So I think we can do this:
```js
var popUp = $element.find('.validation-pop-up-wrapper');
popUp.css({
    height: popUp.height() > 0 ? popUp.height() : 'auto'
});
```
I'm not sure which of the 3 height methods is the exact one we need, I didn't test this, but I think it's worth experimenting and see if we can make this 2-3 lines of code nicer.